code/scripting/api/objs/shipwepselect.cpp

```diff
@@ -189,19 +189,24 @@ ADE_INDEXER(l_Loadout_Amount,
 	idx--; // Convert from Lua
 	if (ADE_SETTING_VAR) {
 		ship_info* sip = &Ship_info[current.getBank()->ship_class];
-		weapon_info* wip = &Weapon_info[current.getBank()->wep[idx]];
-		int capacity = 0;
-		//Calculate max capacity of the weapon in the current bank
-		if (wip->subtype == WP_LASER) {
-			capacity = 1;
+		int widx = current.getBank()->wep[idx];
+		if (widx >= 0) {
```

Comment:
Might as well check that it's also less than the number of valid weapons.  `weapon_info_size()` iirc

Reply:
Yeah might as well. Done.

Reply:
Do I need more coffee or should that be an `&&`?

Reply:
It is apparently I that needs more coffee. It should be `&&`.